### PR TITLE
Add examples folder and HTML page

### PR DESCRIPTION
**Pull request by msssk**
The `examples/index.html` page can be used for all examples.

To create an example page:
* create a TypeScript file in the `examples` folder that creates a projector and appends it to the document.
* add a link to it in `ul#links` in `index.html`

Run `grunt dev` and open `_build/examples/index.html` in your browser. You should see a list of links to test pages.

### CODE REVIEW

**Comment by msssk** [comment]
An enhancement to consider:

Add `paths` config to `tsconfig.json` so that in example files we can use package imports, instead of relative imports, e.g.:

```typescript
import Grid from 'dgrid/Grid';
// instead of
import Grid from '../../src/Grid';
```

P.S. I would have done this but I have no idea how `paths` works in TS, since in my experience it mostly doesn't.

File: Gruntfile.js [hunk 1]
```diff
@@ -12,6 +12,12 @@ module.exports = function (grunt) {
                 cwd: 'src',
                 src: 'styles/dgrid.css',
                 dest: '<%= distDirectory %>'
+            },
+            examples: {
```
Comment (pottedmeat) [inline Gruntfile.js:16]: Should this be called staticExampleFiles like in widgets?

Comment (pottedmeat) [reply Gruntfile.js:16]: I also think this needs to be added to the dev task

**Review by pottedmeat — CHANGES_REQUESTED** [review]

File: examples/index.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	<style>
+		body {
+			font-family: Myriad, Tahoma, Helvetica, Arial, sans-serif;
+		}
+	</style>
+</head>
+
+<body>
+	<ul id="links">
+		<li><a href="?module=pagination">Pagination</a></li>
```
Comment (pottedmeat) [inline examples/index.html]: Should just comment this out for now

File: Gruntfile.js (hunk 1 above)
Comment (msssk) [reply Gruntfile.js:16]: > Should this be called staticExampleFiles like in widgets?

I think we're fine inlining static values that are specific to the dgrid project, as is already being done just above for `devStyles.src` and `distStyles.src`.

Comment (msssk) [reply Gruntfile.js:16]: > I also think this needs to be added to the dev task

Do we also want to include examples in dist?

**Review by pottedmeat — APPROVED** [review]

**Pull request merged**